common/src/main/java/org/opensearch/ml/common/MLConfig.java

```diff
@@ -42,55 +42,93 @@ public class MLConfig implements ToXContentObject, Writeable {
     @Setter
     private String type;
 
+    @Setter
+    private String configType;
+
     private Configuration configuration;
+    private Configuration mlConfiguration;
     private final Instant createTime;
     private Instant lastUpdateTime;
+    private Instant lastUpdatedTime;
 
     @Builder(toBuilder = true)
-    public MLConfig(String type, Configuration configuration, Instant createTime, Instant lastUpdateTime) {
+    public MLConfig(
+        String type,
+        String configType,
+        Configuration configuration,
+        Configuration mlConfiguration,
+        Instant createTime,
+        Instant lastUpdateTime,
+        Instant lastUpdatedTime
+    ) {
         this.type = type;
+        this.configType = configType;
         this.configuration = configuration;
+        this.mlConfiguration = mlConfiguration;
         this.createTime = createTime;
         this.lastUpdateTime = lastUpdateTime;
+        this.lastUpdatedTime = lastUpdatedTime;
     }
 
     public MLConfig(StreamInput input) throws IOException {
         this.type = input.readOptionalString();
+        this.configType = input.readOptionalString();
         if (input.readBoolean()) {
             configuration = new Configuration(input);
         }
+        if (input.readBoolean()) {
+            mlConfiguration = new Configuration(input);
+        }
         createTime = input.readOptionalInstant();
         lastUpdateTime = input.readOptionalInstant();
+        lastUpdatedTime = input.readOptionalInstant();
     }
 
     @Override
     public void writeTo(StreamOutput out) throws IOException {
         out.writeOptionalString(type);
+        out.writeOptionalString(configType);
```

Comment:
Should we consider OS versions ?

Reply:
Yeah I think so. There are going to be like completely new fields now. Not replacing the old ones